azurerm/internal/services/storage/storage_account_resource_test.go

```diff
@@ -715,6 +716,88 @@ func TestAccStorageAccount_hnsWithPremiumStorage(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMStorageAccount_azureFilesIdentityBasedAuthentication(t *testing.T) {
+	data := acceptance.BuildTestData(t, "azurerm_storage_account", "test")
+	r := StorageAccountResource{}
+
+	data.ResourceTest(t, r, []resource.TestStep{
+		{
+			Config: r.basic(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.azureFilesIdentityBasedAuthenticationAD(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		{
+			Config: r.azureFilesIdentityBasedAuthenticationADUpdate(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		// this test step requires a AAD Domain Service exists in the subscription, Else it returns error
+		{
+			Config: r.azureFilesIdentityBasedAuthenticationAADDS(data),
+			Check: resource.ComposeTestCheckFunc(
+				check.That(data.ResourceName).ExistsInAzure(r),
+			),
+		},
+		data.ImportStep(),
+		// the serivce will update in AAD DS after returning response. Issue opened:https://github.com/Azure/azure-rest-api-specs/issues/11272
+		{
+			PreConfig: func() { time.Sleep(3 * time.Minute) },
```

Comment:
In your testing has a 3 minute wait been sufficient every time? We may need to mitigate this in the resource with resource.StateChangeConf.WaitForState().